torch/distributed/_functional_collectives_impl.py

```diff
@@ -43,9 +57,10 @@ def wait(self):
     def decrement_live_tensor(self, ptr):
         self.cleanup_count -= 1
         if self.cleanup_count == 0:
-            self.cleanup()
+            self.wait()
         else:
-            if data_ptr_to_work.get(ptr, None) == self:
+            self.ptr_alias_count[ptr] -= 1
+            if self.ptr_alias_count[ptr] < 1 and data_ptr_to_work.get(ptr, None) == self:
```

Comment:
is there a case where alias count < 1 but the pointer isn't self?  would that be a bug (assert instead?)